lib/gates_def.h

```diff
@@ -26,6 +26,33 @@
 
 namespace qsim {
 
+enum GateKind {
+  kGateId1 = 0, // one-qubit Id
+  kGateHd,      // Hadamard
+  kGateT,       // T
+  kGateX,       // X
+  kGateY,       // Y
+  kGateZ,       // Z
+  kGateX2,      // sqrt(X)
+  kGateY2,      // sqrt(Y)
+  kGateRX,      // X-rotation
+  kGateRY,      // Y-rotation
+  kGateRZ,      // Z-rotation
+  kGateRXY,     // XY-rotation (rotation around arbitrary axis in the XY plane)
+  kGateHZ2,     // pi / 2 rotation around the X + Y axis
+  kGateS,       // S
+  kGateId2,     // two-qubit Id
+  kGateCZ,      // CZ
+  kGateCNot,    // CNOT (CX)
+  kGateIS,      // iSwap
+  kGateFS,      // fSim
+  kGateCP,      // control phase
+  kGateDecomp,  // single qubit gate from Schmidt decomposition
+};
+
+template <typename fp_type>
+using GateQSim = Gate<fp_type, GateKind>;
```

Comment:
Personally, `GateQSim` doesn't help me distinguish between this and the base `Gate` type (aren't all gates qsim gates?) Maybe `DefaultGate` or similar would be clearer, what do you think?

Reply:
Basically, GateQSim refers to the gates implemented in qsim. Gate is a generic gate. The idea is to make it easier to use qsim with external gate sets, which differ from the gate set implemented in qsim.

Reply:
Okay, in the context of external gate sets that naming makes sense. To clarify this meaning for users, could you add a comment in the code here?

Reply:
Done.